[12.0][MIG] pos picking load

legalsylvain opened the pull request:
This PR migrates pos_picking_load from V10 to V12. (The last commit : https://github.com/OCA/pos/commit/229c5da412dbcba3eaf188f4dda899ca210e4709)

pos_picking_load allow to load in the point of sale, confirmed but non delivered out pickings.
![load_picking_01_load_button](https://user-images.githubusercontent.com/3407482/60737315-9cea8900-9f5a-11e9-9ab2-c14ad577c1c8.png)

The list of picking is displayed in the point of sale.

![load_picking_02_picking_list](https://user-images.githubusercontent.com/3407482/60737320-9f4ce300-9f5a-11e9-9dc1-b8c8f601c239.png)


When selected, it generates a pos order based on the stock moves , that can be paid.

![load_picking_04_pos_order](https://user-images.githubusercontent.com/3407482/60737359-ca373700-9f5a-11e9-83ef-5caed1ea4a7d.png)


See the readme part for a more complete description.

CC : @StefanRijnhart, @chienandalu 


Thanks for your review. 

#GRAPOCA

legalsylvain committed:
[ADD] new module pos_picking_load (#183)

[ADD] new module pos_picking_load

oca-transbot committed:
OCA Transbot updated translations from Transifex

StefanRijnhart committed:
[IMP] Link new picking to original sale order
[ADD] Tests

StefanRijnhart committed:
[ADD] Make pos session available when selecting pickings

StefanRijnhart committed:
[IMP] API consistency

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

legalsylvain committed:
[FIX] picking lines quantity are correcty loaded
[ADD] parameter to load a custom quantity of picking in PoS
[IMP] prepare function to let possibility to users to overload search function

oca-travis committed:
[UPD] Update pos_picking_load.pot

oca-travis committed:
[UPD] Update pos_picking_load.pot

legalsylvain committed:
[MIG][10.0] pos_picking_load.
[PORT] Apply new 10.0 syntax ;
[REF] Apply recent OCA convention (new readme, etc.) ;
[PORT] make the module depend on 'sale' module
[REF] remove obsolete file
[IMP] point of sale UI
[ADD] prepare function for picking_line

legalsylvain committed:
[MIG][12.0] pos_picking_load (from 10.0)
[MIG][12.0] pos_picking_load_partner_name (from 8.0)
[FIX] remove obsolete commented code
[FIX] chrome / chromium patch. courtesy @kirca
[FIX] reimplement search picking feature
[FIX] replace min_date by scheduled_date
[UPD] translation

kirca commented on pos_picking_load/tests/test_pos_picking_load.py:
```diff
@@ -7,19 +6,22 @@
         })
         sale_order.action_confirm()
-        self.assertTrue(sale_order.picking_ids)
+        # self.assertTrue(sale_order.picking_ids)
```
Why skip this check?

kirca commented on pos_picking_load/tests/test_pos_picking_load.py:
```diff
@@ -75,7 +79,9 @@ def test_pos_picking_load(self):
         self.assertTrue(pos_order)
         self.assertEqual(pos_order.origin_picking_id.state, 'cancel')
-        self.assertIn(pos_order.picking_id, sale_order.picking_ids)
+        # self.assertIn(pos_order.picking_id, sale_order.picking_ids)
```
Here too. I think if it is not valid anymore it is better to just remove it then commenting it out.

kirca reviewed (commented):
Just added some comments, otherwise looks good. Quite nice module so thanks for initially contributing and maintaining it :).

legalsylvain commented:
Hi @kirca. thanks for your review. I updated my code. Indeed, the commented code is obsolete. I removed it.
Did you tested it functionnaly on runbot ? 
regards

kirca commented on pos_picking_load/static/src/xml/pos_picking_load.xml:
```diff
@@ -0,0 +1,85 @@
+                </section>
+                <section class="full-content">
+                    <div class="window">
+                        <section class="subwindow collapsed">
```
This doesn't seem to be needed, it pushes the selection list down.
![pos_picking_load](https://user-images.githubusercontent.com/4568032/60981828-5b315800-a337-11e9-9aa0-3ed9459167ff.png)

legalsylvain replied:
Hi @kirca ! 

Well, I copy pasted windows present in Odoo. (partner selection).

On runbot, in my computer, it works like a charm.

![pos_picking_load](https://user-images.githubusercontent.com/3407482/60983800-e6f8b380-a33a-11e9-9223-8a4cb038456b.gif)

So, I guess it's browser trouble don't you think ? 

for my Side, Mozilla Firefox for Ubuntu, 67.0.4 (64 bits)

do you face the same problem, when you select a customer ?

thanks !

kirca replied:
Indeed, issue is with Chrome. I did some digging and found commit where there is fix for it in the customer selection screen https://github.com/odoo/odoo/commit/4ff34a98bcefa96c908baae1646a5af8acd21913#diff-401e9bb2837c2ac0b281c1c8a9c195ceR440 . Just need to add "collapsed" class next to "subwindow-container".

Ubuntu 16.04
Chrome - Version 72.0.3626.119 (Official Build) (64-bit)
Firefox - 67.0.4 (64-bit)

kirca commented on pos_picking_load/static/src/xml/pos_picking_load.xml:
```diff
@@ -0,0 +1,85 @@
+                        Select
+                    </span>
+                    <span class="searchbox">
+                        <input placeholder="Search Picking" />
```
The search doesn't seem to be handled in JS and it is not doing anything.

legalsylvain replied:
Oh yes, my bad. Forgot to port this section.

kirca commented:
Hi @legalsylvain , test on runbot revealed some issues. Please check my comments.

legalsylvain commented:
Hi @kirca  thanks a lot for your review, and your help regarding Chrome trouble.
- I fixed, the chrome bug, as you proposed ; 
- I restored search feature ; 
- I renamed min_date by scheduled_date for picking model ; 
- I added an extra very light module ``pos_picking_load_partner_name``. This module store the name of the partner on each picking, and change the search feature in pos_picking_load. the reason is that the domain [('partner_id', 'ilike', my_query)] is very slow on a huge database with millions of pickings. (~10 seconds) with that extra optional module, the search is performed in < 0.3 second.

regards.

Should be OK now.

kirca commented:
Hi @legalsylvain , all changes look good. Also tested again on runbot and everything is working great. Nice job!

rafaelbn commented:
Hi @kirca , is all is OK, aprrove the PR please. Thanks!

kirca reviewed (approved):
@rafaelbn all looks good to me

legalsylvain commented:
@rafaelbn, @chienandalu : any update ?
regards.

legalsylvain commented:
/ocabot merge

chienandalu reviewed (approved):
This merge failed on tests. Maybe you can rebase (and squash some of the commits) and see if we can merge already. Otherwise good for me

chienandalu commented:
travis not happy :/

legalsylvain commented:
Hi all. After investigation, it seems that the error is present if ``pos_picking_delayed`` is installed. 
I made a PR for that purpose. (picking is not created, so procurement group test fails).
Could you review this trivial patch ? 

Thanks ! 
https://github.com/OCA/pos/pull/486

chienandalu commented:
Done :)

legalsylvain commented:
/ocabot merge patch

gaikaz commented on pos_picking_load/models/sale_order.py:
```diff
@@ -0,0 +1,13 @@
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    final_pos_order_id = fields.Many2one(
+        string='Final PoS Order', comodel_name='pos.order', readonly=True,
+        help="This Sale Order has beend replaced by this PoS Order")
```
I can't seem to find, how this field is ever set. I suppose it was meant to be set in `_handle_orders_with_original_picking` method of `pos.order`?

gaikaz reviewed (commented):
Very useful and great modules. :+1: 

I know this is already approved (though yet to merged for some issues with tests), but still:
In general **LGTM and functionally does what it says**, except for that one field on Sale Order.

Also, it would be nice to have it log a note (comment) in the canceled picking chatter, with a link to a new picking that replaces the "origin" one.
But maybe that's for me to implement as a custom addition :smile: Or at least not in this PR.

gaikaz commented:
@legalsylvain Do you by any chance have a planned time to return to this PR? Maybe we just need to nudge someone a little to just merge it already?

Asking, because we kind of need this for v13. I could probably do a PR for migration to v13, but it would depend on this one.

legalsylvain committed:
[FIX] pos_picking_delayed: include demo data

legalsylvain commented:
Hi @gaikaz 

This should be OK now.

/ocabot merge patch

Kind regards.
